.pre-commit-config.yaml

```diff
@@ -21,7 +21,7 @@ repos:
       exclude: _.*_dict.py
 - repo: https://github.com/astral-sh/ruff-pre-commit
   # Ruff version.
-  rev: v0.4.7
+  rev: v0.5.1
```

Comment:
Shouldn't this be done by the bot?

Reply:
Yes, I think so. We can leave the version bump for later ("monthly"), and just apply the new Ruff 0.5.0 rules for now.

Reply:
That's ok, we can also do this now... less work for the bot 😀

Reply:
Except in the meantime I always get a thousand pre-commit PRs because the `pyproject.toml` ruff version that I dev with doesn't match the pre-commit version.

Reply:
Can you bump the `pyproject.toml` ruff version too, please?

Reply:
Note that Ruff versions are defined both in the pre-commit configuration:
https://github.com/pydicom/pydicom/blob/57367af501cd3648564761ef12d8e15438e924d8/.pre-commit-config.yaml#L23-L24
and the project configuration:
https://github.com/pydicom/pydicom/blob/57367af501cd3648564761ef12d8e15438e924d8/pyproject.toml#L53

These versions are updated separately, see this pull request #2098 and previous pull request #2086. This doesn't sound right.

Reply:
Right. I think we can remove all dependencies that are installed and managed by `pre-commit` from `pyproject.toml`.
`pre-commit` does not (and will not by design) use `pyproject.toml`, and as it installs all tools in a sandbox (e.g. a virtual environment), these need not to be added to the dev environment.

Reply:
Fair point. I have pre-commit installed as a hook, so I'm used to run it automatically, but it sure can be a bit intrusive. I'm ok to just leave it in - in the end, it is only there as a convenience for developers, and not used in the CI.